### DISCUSSION TITLE
Dropping unmatched character ️ (U+fe0f) in '⌨️'

### DISCUSSION THREAD

**Discussion by sarmong** [polybar/polybar#2871] [Q&A]
I have keyboard emoji in my module output to polybar - ⌨️.

It is showing correctly in polybar - 
![image](https://user-images.githubusercontent.com/42828375/205448846-d19f6566-f1f7-42e7-bfc1-677e87cd464d.png)

However, I still get the following warning
![image](https://user-images.githubusercontent.com/42828375/205448862-e8662efd-f2e2-4592-99fc-8eef2fd95824.png)

These are my font settings

```
font-0 = fixed:pixelsize=10;1polybar
font-1 = unifont:fontformat=truetype:size=8:antialias=false;0
font-2 = NotoColorEmoji:scale=10
```

P.S. Other emojis work fine. I've noticed that this particular emoji takes two columns in the terminal, so perhaps that makes it different from others and causes error?

**Comment by patrick96** [comment]
The problem is that polybar does not fully support emojis. It only supports single character emojis. What you have here is the keyboard character together with something called a ["Emoji variation selector"](https://emojipedia.org/variation-selector-16).

In this case it takes the keyboard character (`U+2328`), which is not an emoji, and turns it into an emoji.

**Reply by sarmong** [reply]
Thanks for the explanation.

What is weird is that other emojis also consist of 2 characters, but apparently behave differently. For example, in neovim when I move a cursor to a keyboard emoji, I can go to column 1, which shows the cursor on whole emoji ![image](https://user-images.githubusercontent.com/42828375/209567939-a43ef54e-0a9e-4954-9597-f53ea80bef5b.png) and also go to column 2  ![image](https://user-images.githubusercontent.com/42828375/209567994-84d7ffb2-7277-4336-96c2-b479872dbd0b.png), which highlights this variant selector.

For most other emojis, I can _only_ go to column 2  ![image](https://user-images.githubusercontent.com/42828375/209568083-a0ab3d5c-422e-464e-a255-4c3859a2d543.png). So looks like this is also a two-character emoji.


Also, I wonder why if I remove this variant selector, I am still shown an emoji in neovim.
And if I put the unicode character into polybar `format-prefix = "⌨"`, I still get emoji in polybar.

**Reply by patrick96** [reply]
The thing you're seeing in polybar is not actually an emoji. You are seeing the keyboard character `⌨` (`U+2328`) which is not the same as the keyboard emoji `⌨️` (`U+2328` `U+FE0F`).

Some fonts may choose to render them the same, but for me they look like this in the terminal:

![2022-12-26-224405_905x95_scrot](https://user-images.githubusercontent.com/2452038/209584906-47450ae0-2bf5-48bd-987b-55d80622c816.png)
